pkg/controllers/deprovisioning/helpers.go

```diff
@@ -28,10 +28,12 @@ import (
 	pscheduling "github.com/aws/karpenter-core/pkg/controllers/provisioning/scheduling"
 	"github.com/aws/karpenter-core/pkg/controllers/state"
 	"github.com/aws/karpenter-core/pkg/scheduling"
-	"github.com/aws/karpenter-core/pkg/utils/node"
+	nodeutils "github.com/aws/karpenter-core/pkg/utils/node"
+	"github.com/aws/karpenter-core/pkg/utils/pod"
 
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/utils/clock"
```

Comment:
It feels like `simulateScheduling` is a core part of controllers so it feels weird that this is in the helpers file

Reply:
It's a helper function used in any deprovisioner that implements replacement. We don't use it in the main deprovisioning loop, so I think it's best here. WDYT?